proposals/gc/MVP.md

```diff
@@ -597,36 +597,33 @@ In particular, `ref.null` is typed as before, despite the introduction of `none`
 
 Casts work for both abstract and concrete types. In the latter case, they test if the operand's RTT is a sub-RTT of the target type.
 
-* `ref.test null? <heaptype>` checks whether a reference has a given heap type
-  - `ref.test null? ht : [(ref null ht')] -> [i32]`
-    - iff `ht <: tht` and `ht' <: tht` where `tht` is a common super type
-  - if `null?` is present, returns 1 for null, otherwise 0
-
-* `ref.cast null? <heaptype>` tries to convert to a given heap type
-  - `ref.cast null? ht : [(ref null ht')] -> [(ref null2? ht)]`
-    - iff `ht <: tht` and `ht' <: tht` where `tht` is a common super type
-    - and `null? = null2?`
+* `ref.test <reftype>` tests whether a reference has a given type
+  - `ref.test rt : [rt'] -> [i32]`
+    - iff `rt <: trt` and `rt' <: trt` for some `trt`
+  - if `rt` contains `null`, returns 1 for null, otherwise 0
+
+* `ref.cast <reftype>` tries to convert a reference to a given type
+  - `ref.cast rt : [rt'] -> [rt]`
+    - iff `rt <: trt` and `rt' <: trt` for some `trt`
   - traps if reference is not of requested type
-  - if `null?` is present, a null operand is passed through, otherwise traps on null
-  - equivalent to `(block $l (param anyref) (result (ref null? ht)) (br_on_cast null? ht $l) (unreachable))`
+  - if `rt` contains `null`, a null operand is passed through, otherwise traps on null
+  - equivalent to `(block $l (param anyref) (result rt) (br_on_cast $l rt) (unreachable))`
```

Comment:
I think, this isn't true in all cases. For externref types or funcref types `anyref` would not be compatible.
is there a way to make it look nice while also applying to funcref types? Otherwise we could add a `(for input values of the anyref type hierarchy)` or a similar restriction to this line.

Reply:
Ah, indeed, this ceased to hold when we split the type hierarchy.